src/session.rs

```diff
@@ -328,20 +337,18 @@ impl Session {
             }
         }
     }
-
-    pub(crate) async fn with_capabilities(
+    pub(crate) async fn with_capabilities_and_connector(
         webdriver: &str,
-        mut cap: webdriver::capabilities::Capabilities,
+        cap: &webdriver::capabilities::Capabilities,
+        connector: C,
     ) -> Result<Client, error::NewSessionError> {
         // Where is the WebDriver server?
         let wdb = webdriver.parse::<url::Url>();
-
         let wdb = wdb.map_err(error::NewSessionError::BadWebdriverUrl)?;
-
         // We want a tls-enabled client
-        let client = hyper::client::Client::builder()
-            .build::<_, hyper::Body>(hyper_tls::HttpsConnector::new());
+        let client = hyper::Client::builder().build::<_, hyper::Body>(connector);
 
+        let mut caps = cap.to_owned();
```

Comment:
We could just make this
```rust
let mut cap = cap.to_owned();
```
and that way much of the code below can stay the same.